src/Evolution/Systems/Cce/AnalyticSolutions/GaugeWave.hpp

```diff
@@ -0,0 +1,272 @@
+ * tested by the `Cce::Solutions::BouncingBlackHole` solution.
+ */
+struct GaugeWave : public SphericalMetricData {
+  struct ExtractionRadius {
+    using type = double;
+    static constexpr Options::String help{
+        "The extraction radius of the spherical solution"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Mass {
+    using type = double;
+    static constexpr Options::String help{
+        "The mass of the Schwarzschild solution."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Frequency {
+    using type = double;
+    static constexpr Options::String help{
+        "The frequency of the oscillation of the gauge wave."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Amplitude {
+    using type = double;
+    static constexpr Options::String help{
+      "The amplitude of the gauge wave."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct PeakTime {
+    using type = double;
+    static constexpr Options::String help{
+        "The time of the peak of the Gaussian envelope."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Duration {
+    using type = double;
+    static constexpr Options::String help{
+        "The characteristic duration of the Gaussian envelope."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  using options = tmpl::list<ExtractionRadius, Mass, Frequency, Amplitude,
+                             PeakTime, Duration>;
+
+  static constexpr Options::String help = {
+      "Analytic solution representing worldtube data for a pure-gauge "
+      "perturbation near a Schwarzschild metric in spherical coordinates"};
+
+  WRAPPED_PUPable_decl_template(GaugeWave);  // NOLINT
+
+  explicit GaugeWave(CkMigrateMessage* /*unused*/) noexcept {}
+
+  // clang doesn't manage to use = default correctly in this case
+  // NOLINTNEXTLINE(modernize-use-equals-default)
+  GaugeWave() noexcept {}
+
+  GaugeWave(double extraction_radius, double mass, double frequency,
+            double amplitude, double peak_time, double duration) noexcept;
+
+  std::unique_ptr<WorldtubeData> get_clone() const noexcept override;
+ private:
+  double coordinate_wave_function(double time) const noexcept;
+
+  double du_coordinate_wave_function(double time) const noexcept;
+
+  double du_du_coordinate_wave_function(double time) const noexcept;
+
+ protected:
+  /// A no-op as the gauge wave solution does not have substantial
+  /// shared computation to prepare before the separate component calculations.
+  void prepare_solution(const size_t /*output_l_max*/,
+                        const double /*time*/) const noexcept override {}
+
+  /*!
+   * \brief Compute the spherical coordinate metric from the closed-form gauge
+   * wave metric.
+   *
+   * \details The transformation of the ingoing Eddington-Finkelstein coordinate
+   * produces metric components in spherical coordinates:
+   *
+   * \f{align*}{
```

Comment:
I would just reference the equation this comes from, i.e., Eq. (149)

Reply:
I'll add the reference, but I think it's probably best to keep the explicit form here as well, just because there are some alterations in the precise form to e.g. minimize division operations to improve anticipated efficiency, and it's easiest for verification if the implementation is identical to the tex to compare with.